server/src/main/java/io/crate/metadata/SimpleReference.java

```diff
@@ -221,6 +221,9 @@ public Map<String, Object> toMapping() {
                 mapping.put(TypeParsers.DOC_VALUES, Boolean.toString(hasDocValues));
             }
         }
+        if (defaultExpression != null) {
+            mapping.put("default_expr", defaultExpression.toString(Style.UNQUALIFIED));
+        }
```

Comment:
Here we don't take to account potential necessity of casting like we do in analyzer 
https://github.com/crate/crate/blob/master/server/src/main/java/io/crate/analyze/AnalyzedTableElements.java#L440-L448

It's a component which creates SimpleReference to put a proper expression. In other PR-s, when we create List<References> out of AnalyzedColumnDefinition-s, we will provide a proper Reference with a proper expression (with casts included if necessary)